src/Razor/src/Microsoft.AspNetCore.Razor.Language/Legacy/CodeBlockEditHandler.cs

```diff
@@ -103,7 +103,7 @@ internal static bool IsAcceptableInsertion(SourceChange change)
         // Internal for testing
         internal static bool ContainsInvalidContent(SourceChange change)
         {
-            if (change.NewText.IndexOfAny(new[] { '{', '}' }) >= 0)
+            if (change.NewText.IndexOfAny(new[] { '{', '}', '@', '<', '*', }) >= 0)
```

Comment:
@NTaylorMullen - looking for your advice on this, plus any ideas you have if I'm missing tests. I had to add all of these various characters to make partial-parsing work but other than that it was straightforward.

I considered making these conditional on the version, but functions doesn't get much usage outside of 3.0 scenarios anyway....

Reply:
Played around with this a bit and it looks like it doesn't work exactly how you'd expect. For instance if you have 
```
@functions  {
    
    |

    @<p></p>
}
```

Typing anything at the pipe wont end up hitting the codeblock edit handler. Seems as though bits don't get classified properly when the functions block already has a template in it.